src/Microsoft.OpenApi/Models/OpenApiReference.cs

```diff
@@ -83,7 +83,7 @@ public string ReferenceV3
                     return Id;
                 }
 
-                return "#/components/" + Type.GetDisplayName() + "/" + Id;
+                return "#/components/" + Type.Value.GetDisplayName() + "/" + Id;
```

Comment:
Here and below, we unwrap the value from the nullable type to support the compiler in being able to resolve the `ReferenceType` in our generic `GetDisplayName` method.

Reply:
could we have the extension methods bind to the nullable type instead?